phd-tests/framework/src/test_vm/metrics.rs

```diff
@@ -0,0 +1,200 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+use std::net::SocketAddr;
+use std::time::Duration;
+
+use dropshot::{
+    endpoint, ApiDescription, ConfigDropshot, HttpError, HttpResponseCreated,
+    HttpServer, HttpServerStarter, RequestContext, TypedBody,
+};
+use omicron_common::api::internal::nexus::{
+    ProducerEndpoint, ProducerKind, ProducerRegistrationResponse,
+};
+use oximeter::types::ProducerResults;
+use slog::{Drain, Logger};
+use tokio::sync::watch;
+use tracing::trace;
+use uuid::Uuid;
+
+// Re-registration interval for tests. A long value here helps avoid log spew
+// from Oximeter, which will re-register after about 1/6th of this interval
+// elapses.
+const INTERVAL: Duration = Duration::from_secs(300);
+
+fn test_logger() -> Logger {
+    let dec = slog_term::PlainSyncDecorator::new(slog_term::TestStdoutWriter);
+    let drain = slog_term::FullFormat::new(dec).build().fuse();
+    Logger::root(drain, slog::o!("component" => "fake-cleanup-task"))
```

Comment:
nit: I might pick a more descriptive component name; maybe `phd-oximeter-consumer`?

Reply:
definitely, this was not a great holdover name :) went with `phd-oximeter-consumer`.